jgrapht-core/src/main/java/org/jgrapht/alg/planar/BoyerMyrvoldPlanarityInspector.java

```diff
@@ -867,7 +867,9 @@ private boolean findPathDfs(Node start, Edge startPrev, Predicate<Node> canGo, P
                  * The iteration is performed in the reverse order since the infos are pushed on
                  * the stack and therefore will be processed in the again reverse order
                  */
-                for (Iterator<Edge> iterator = info.current.embedded.reverseIteratorFrom(info.prevEdge); iterator.hasNext(); ) {
+                Iterator<Edge> iterator =
+                    info.current.embedded.wrappingIterator(info.prevEdge, true);
+                while (iterator.hasNext()) {
```

Comment:
@Toptachamann is the behavior of the wrappingIterator correct in this place?

Reply:
Good. I also checked it and the tests passed but wanted your confirmation since your the author and know it best.